pages/index.tsx

```diff
@@ -8,18 +8,28 @@ import LoginWithSMS from '../components/LoginWithSMS';
 import { LoginMethod } from '../lib/types';
 import LoginEntryPoint from '../components/LoginEntryPoint';
 
-const REDIRECT_URL_BASE = process.env.USE_VERCEL == 'true' ?
-  `https://${process.env.VERCEL_URL}/api/authenticate_magic_link` :
-  'http://localhost:3000/api/authenticate_magic_link';
+// Set the URL base for redirect URLs. The three cases are as follows:
+// 1. Running locally via `vercel dev`; VERCEL_URL will contain localhost, but will not be https.
+// 2. Deploying via Vercel; VERCEL_URL will be generated on runtime and use https.
+// 3. Running locally via `npm run dev`; VERCEL_URL will be undefined and the app will be at localhost.
+let REDIRECT_URL_BASE = '';
 
+if (process.env.VERCEL_URL != undefined &&
+    process.env.VERCEL_URL.includes('localhost')) {
+  REDIRECT_URL_BASE = 'http://localhost:3000';
+} else if (process.env.VERCEL_URL != undefined) {
+  REDIRECT_URL_BASE = `https://${process.env.VERCEL_URL}`;
+} else {
+  REDIRECT_URL_BASE = 'http://localhost:3000';
+}
```

Comment:
This is a little ugly, but as I was working on the Vercel-only version of this example app today I realized that it'd be nice to support `npm run dev`, `vercel dev`, and deploy with Vercel in both. 

Open to a rewrite of this if block, but it made sense in my head so that usually is enough for me :)

Reply:
I don't immediately see a way to clean it up besides tiny things like using `?.includes`, so 🤷 . The comment makes it clear what cases this is supposed to handle, so I'd feel comfortable adding a case in the future and knowing how to test it. Maybe it's worth hiding this in a function like `resolveRedirectURLBase`?

Reply:
Moved to optional chaining here: 790cfcb

My limit has generally been, "Up to three if's" before rewriting as a function. Since we're at that limit my inclination is to leave as-is and let future `chris-stytch` worry about it if we add other cases.

Reply:
lol that's a reasonable rule. I don't feel strongly about it, so 🚀